includes/class-wc-adapter.php

```diff
@@ -0,0 +1,682 @@
+			ConfigurationInterface::MERCHANT_TOKEN         => $this->gateway->merchant_token,
+			ConfigurationInterface::PAYEE_ID               => $this->gateway->payee_id,
+			ConfigurationInterface::PAYEE_NAME             => get_bloginfo( 'name' ),
+			ConfigurationInterface::MODE                   => 'yes' === $this->gateway->testmode,
+			ConfigurationInterface::AUTO_CAPTURE           => 'yes' === $this->gateway->auto_capture,
+			ConfigurationInterface::SUBSITE                => $this->gateway->subsite,
+			ConfigurationInterface::LANGUAGE               => $this->gateway->culture,
+			ConfigurationInterface::SAVE_CC                => 'yes' === $this->gateway->save_cc,
+			ConfigurationInterface::TERMS_URL              => $this->gateway->terms_url,
+			ConfigurationInterface::REJECT_CREDIT_CARDS    => 'yes' === $this->gateway->reject_credit_cards,
+			ConfigurationInterface::REJECT_DEBIT_CARDS     => 'yes' === $this->gateway->reject_debit_cards,
+			ConfigurationInterface::REJECT_CONSUMER_CARDS  => 'yes' === $this->gateway->reject_consumer_cards,
+			ConfigurationInterface::REJECT_CORPORATE_CARDS => 'yes' === $this->gateway->reject_corporate_cards,
+		);
+	}
+
+	/**
+	 * Get Platform Urls of Actions of Order (complete, cancel, callback urls).
+	 *
+	 * @param mixed $order_id
+	 *
+	 * @return array
+	 */
+	public function getPlatformUrls( $order_id ) {
+		$order = wc_get_order( $order_id );
+
+		if ( $this->gateway->is_new_credit_card ) {
+			return array(
+				PlatformUrlsInterface::COMPLETE_URL => add_query_arg(
+					'action',
+					'swedbank_card_store',
+					admin_url( 'admin-ajax.php' )
+				),
+				PlatformUrlsInterface::CANCEL_URL   => wc_get_account_endpoint_url( 'payment-methods' ),
+				PlatformUrlsInterface::CALLBACK_URL => WC()->api_request_url( get_class( $this->gateway ) ),
+				PlatformUrlsInterface::TERMS_URL    => '',
+			);
+		}
+
+		if ( $this->gateway->is_change_credit_card ) {
+			return array(
+				PlatformUrlsInterface::COMPLETE_URL => add_query_arg(
+					array(
+						'verify' => 'true',
+						'key'    => $order->get_order_key(),
+					),
+					$this->gateway->get_return_url( $order )
+				),
+				PlatformUrlsInterface::CANCEL_URL   => $order->get_cancel_order_url_raw(),
+				PlatformUrlsInterface::CALLBACK_URL => WC()->api_request_url( get_class( $this->gateway ) ),
+				PlatformUrlsInterface::TERMS_URL    => $this->getConfiguration()[ ConfigurationInterface::TERMS_URL ],
+			);
+		}
+
+		return array(
+			PlatformUrlsInterface::COMPLETE_URL => $this->gateway->get_return_url( $order ),
+			PlatformUrlsInterface::CANCEL_URL   => $order->get_cancel_order_url_raw(),
+			PlatformUrlsInterface::CALLBACK_URL => WC()->api_request_url( get_class( $this->gateway ) ),
+			PlatformUrlsInterface::TERMS_URL    => $this->getConfiguration()[ ConfigurationInterface::TERMS_URL ],
+		);
+	}
+
+	/**
+	 * Get Order Data.
+	 *
+	 * @param mixed $order_id
+	 *
+	 * @return array
+	 */
+	public function getOrderData( $order_id ) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method getOrderData() has a Cyclomatic Complexity of 22. The configured cyclomatic complexity threshold is 10.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)